core/src/main/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizer.java

```diff
@@ -19,14 +23,24 @@
  * 1> Optimize the current node with all the rules.
  * 2> Optimize the all the child nodes with all the rules.
  * 3) In case the child node could change, Optimize the current node again.
+ * TODO update comment ^
  */
 @RequiredArgsConstructor
 public class LogicalPlanOptimizer {
 
   private final List<Rule<? extends LogicalPlan>> rules;
 
+  private final OptimizingMode optimizingMode;
```

Comment:
I would suggest removing the Mode.  It's not needed.  
Instead let's create a LogicalPlanOptimizer interface, and a separate SQLLogicalPlanOptimizer (that contains the `create2()` function) and PPLLogicalPlanOptimizer (that contains the old `create()` function.  The Planner can be created with the appropriate PlanOptimizer. 
I might consider also checking for aggregation in the plan and consider using the 'old' pushdown method when aggregation is present in the plan... that would be safer.  But if you're confident that this pushdown works with aggregation, then we can push.